src/attr/_make.py

```diff
@@ -135,10 +138,15 @@ def attrib(
         as-is, i.e. it will be used directly *instead* of calling ``repr()``
         (the default).
     :type repr: a ``bool`` or a ``callable`` to use a custom function.
-    :param bool cmp: Include this attribute in the generated comparison methods
-        (``__eq__`` et al).
+    :param bool eq: If ``True`` (default), include this attribute in the
```

Comment:
My question would be: is the fact that we can't come up with a reason right now worth it to:

1. break the symmetry of `@attr.s()` and `attr.ib()`
2. confusingly  keep an argument around, that doesn't exist in `@attr.s` anymore?

Since `order` mirrors `eq` by default, it's not more arguments to pass and maybe there is at some point _a_ reason to order over a subset but compare over everything? `eq=False, order=True` is invalid ATM and I'd keep that invariant around until someone give me a good use case.

Reply:
I have to think about this, like I said my initial intuition is the other way, but I'll try to convince myself otherwise.

But, for clarity, the other side of the argument you mentioned (and in fact the side mine above is entirely rooted in) is "the symmetry here is false -- these arguments have the same name, but they affect a very different part of the equation" and more broadly, that you can't put cats back in bags.

We can always add the capability to change these individually later if someone compels with a use case, and to your point, the cmp argument is staying no matter what whilst it's supported for backwards compat, so it's a net *non* addition to not add this functionality.

Whereas adding it potentially, again at first glance, adds possibility of mistake for end users to create "non-sensical objects".

And yeah I don't think this has to do with that eq=False order=True combo, the nonsensical objects come about when both *are* enabled.

Just my 0.02 before having morning coffee though...

Reply:
`cmp`is supposed to go ion 2021 though. And I'd be rather unhappy to keep around just for this one case.

I really don't think those objects are really non-sensical?

I mean `eq=False,order=True` Can't happen. Both True and both False make sense. As for `True,False` how about this:

```
@attr.s
class Person:
    id = attr.ib(order=False)
    name = attr.ib()
```

Since we're both not entirely sure, let's try to ask someone else: @glyph?

Reply:
OK *so* -- now that I'm home and actually have time to open a REPL and try using this :) -- I totally misunderstood that the `eq=False, order=True` error applied here too, I thought you were telling me about `attr.s(eq=False, order=True)`, not `attr.ib(eq=False, order=True)` -- so it was definitely the latter that I think is non-sensical.

The person example I think is... "ugly", but someone is going to want to do it anyhow, so yeah not bad enough to me that I'd think you'd not allow it.

So anyways the way this seems to work I *think* makes sense to me now :) so apologies for the slow understanding.

Probably still makes sense to get a third opinion on this point but yeah I think this is probably fine to me too!

Reply:
Cool, thanks for taking the time on your trip to talk it through with me – your instagram had nicer vistas than my crummy code. :)

Reply:
HAH